prdoc/schema_user.json

```diff
@@ -0,0 +1,223 @@
+    "required": [
+      "title",
+      "doc",
+      "crates",
+      "migrations",
+      "host_functions"
+    ],
+    "$defs": {
+      "audience": {
+        "oneOf": [
+          {"const": "Node Dev",
+          "title": "Those who build around the client side code. Alternative client builders, SMOLDOT, those who consume RPCs. These are people who are oblivious to the runtime changes. They only care about the meta-protocol, not the protocol itself."},
+
+          {"const": "Runtime Dev",
+          "title": "All of those who rely on the runtime. A parachain team that is using a pallet. A DApp that is using a pallet. These are people who care about the protocol (WASM), not the meta-protocol (client)."},
+
+          {"const": "Node Operator",
+          "title": "Those who don't write any code and only run code."},
+
+          {"const": "Runtime User",
+          "title": "Token holders who don't run anything but are involved in the network."}
+        ]
+      },
+      "crate": {
+        "type": "object",
+        "additionalProperties": false,
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "semver": {
+            "$ref": "#/$defs/semver"
+          },
+          "note": {
+            "type": "string"
+          }
+        }
+      },
+      "semver": {
+        "type": "string",
+        "enum": [
+          "major",
+          "minor",
+          "patch",
+          "no change"
+        ]
+      },
+      "migration_db": {
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          }
+        },
+        "additionalProperties": false,
+        "required": [
+          "name",
+          "description"
+        ]
+      },
+      "migration_runtime": {
+        "type": "object",
+        "properties": {
+          "pallet": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          }
+        },
+        "additionalProperties": false,
+        "required": [
+          "pallet",
+          "description"
+        ]
+      },
+      "doc": {
+        "type": "object",
+        "description": "You have the the option to provide different description of your PR for different audiences.",
+        "additionalProperties": false,
+        "properties": {
+          "audience": {
+            "description": "The selected audience",
+            "$ref": "#/$defs/audience"
+          },
+          "title": {
+            "type": "string",
+            "description": "Optional title override for the PR and for the current audience"
+          },
+          "description": {
+            "description": "Description of the change",
+            "type": "string"
+          },
+          "notes": {
+            "description": "Optional relevant note",
+            "$ref": "#/$defs/array_of_strings"
+          }
```

Comment:
What is the difference between notes and description?

Reply:
Good point. This is mainly to make prdoc easier to write and to control the rendering better.

The description is a chunk of text. It could end up with a list of notes/notice/heads up in theory but this is not optimum.

The `notes` are optional and managed as an array of strings.
Having `notes` like this allows writing:
```
doc:
  - audience: Node Operator
    description: |
      lorem ipsum
   notes:
      - Make sure to do this
      - Make sure you have that
```

instead of:
```
doc:
  - audience: Node Operator
    description: |
      lorem ipsum
      NOTES:
          - Make sure to do this
          - Make sure you have that
```

That may look like a small difference but it mainly allows controlling the rendering in the templates and not in the content of the prdoc. 

For instance, we may want the notes to be collapsable in the rendered markdown, without having to write those `<details><summary>...</summary></details>` here.

Reply:
Again, over complicating it. Please keep it simple. So, please remove `Notes`.

TBH, I'm not entirely convinced that  yaml was such a great idea any way.